kuwala/core/canvas/src/pages/DataSourcePreview.js

```diff
@@ -0,0 +1,194 @@
+                                            src={ArrowDown}
+                                            style={{width: 16, height: 16}}
+                                        />
+                                        </span>
+                                        <span className={'mr-4'}>
+                                            <img
+                                                src={ListSVG}
+                                                style={{width: 16, height: 16}}
+                                            />
+                                        </span>
+                                        <span className={'font-semibold text-md'}>
+                                            Canvas
+                                        </span>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                    <div className={'flex flex-col bg-white w-5/6 overflow-auto'}>
+                        <table className={'bg-white'}>
+                            <thead>
+                            <tr>
+                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 1</th>
+                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 2</th>
+                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 3</th>
+                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 4</th>
+                                <th className={'bg-kuwala-light-green text-white lowercase px-4 py-2'}>Header 5</th>
+                            </tr>
+                            </thead>
+                            <tbody>
+                                {[...Array(100).keys()].map((e) => {
+                                    console.log(e)
+                                    return (
+                                        <tr>
+                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
+                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
+                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
+                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
+                                            <td className={'text-left px-4 py-2 border border-kuwala-light-green'}>THINGS</td>
+                                        </tr>
+                                    )
+                                })}
+                            </tbody>
+                        </table>
+                    </div>
+                </div>
```

Comment:
I feel the schema on the left and the table data should be two separate render functions.

Reply:
Yup, still work in progress. it will be separated into smaller render functions later on.